test/python/circuit/library/test_nlocal.py

```diff
@@ -868,6 +902,19 @@ def test_circuit_with_numpy_integers(self):
         self.assertEqual(two_np32.decompose().count_ops()["cx"], expected_cx)
         self.assertEqual(two_np64.decompose().count_ops()["cx"], expected_cx)
 
+    @combine(num_qubits=[3, 4, 5, 6, 10])
+    def test_full_vs_reverse_linear(self, num_qubits):
```

Comment:
This test might not be necessary if "full" and "reverse linear" are analytically equivalent. Especially because this might run a bit slow for large numbers of qubits.

Reply:
I still prefer to have this test, since someone may try to change the code of `'full'` or `'reverse_linear'` in the future. 
However, I reduced the number of checks from `num_qubits=[3, 4, 5, 6, 10]` to `num_qubits=[4, 5]`